test-data/unit/check-typeddict.test

```diff
@@ -493,6 +493,26 @@ reveal_type(fun(b))  # N: Revealed type is "builtins.object*"
 [builtins fixtures/dict.pyi]
 [typing fixtures/typing-typeddict.pyi]
 
+[case testTypedDictTempTestName]
+from typing import Any, Type, TypeVar
+from typing_extensions import TypedDict
+
+T = TypeVar('T')
+class C(TypedDict):
+    x: str
+
+def gen() -> Any:
+    return {'x': 1}
+def foo(cls: Type[T]) -> T:
+    return gen()
+def fun(c: C):
+    pass
+
+fun(foo(C))
```

Comment:
The call `foo(C)` should actually generate an error (see my other comment). For example, if the call would make sense, we should handle cases like these, which probably won't work correctly right now:

```py
d = foo(C)
d['x'] = 1  # Should be ok
d['y'] = 1  # Should be an error
d['x'] = 'x'  # Should be an error
```